### PR TITLE
samples: openthread: enable the Link Metrics and CSL features for nRF53

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
To enable Thread v1.2 with the Link Metrics and CSL features
set -DCONF_FILE=prj_thread_1_2.conf. It apart from setting
-DCONFIG_OPENTHREAD_THREAD_VERSION_1_2=y, applies child image
additional configuration for network core build for dual-core
architecture like nRF53.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

### CODE REVIEW

**Review by wiba-nordic — APPROVED** [review]
Looks good.

File: samples/openthread/cli/child_image/802154_rpmsg_thread_1_2.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
```
Comment (edmont) [inline samples/openthread/cli/child_image/802154_rpmsg_thread_1_2.conf:2]: Please change dates for the new files.

**Review by edmont — APPROVED** [review]
LGTM, but maybe this change is big enough to add some note to the release changelog?

**Comment by tomaszkob89** [comment]
We should add prj_thread_1_2.conf also to coprocessor samples

**Review by wiba-nordic — APPROVED** [review]
Changelog looks good as well.

**Review by canisLupus1313 — APPROVED** [review]

**Review by tomaszkob89 — APPROVED** [review]

**Pull request merged**